dart/lib/src/sentry_span_interface.dart

```diff
@@ -0,0 +1,33 @@
+import '../sentry.dart';
+
+abstract class ISentrySpan {
+  ISentrySpan startChild(
+    String operation, {
+    String? description,
+  });
+
+  void setTag(String key, String value);
+
+  void removeTag(String key);
+  // exposed to Transaction too but it does not get sent to the backend since transactions dont have the data bag
+  void setData(String key, dynamic value);
+// exposed to Transaction too but it does not get sent to the backend since transactions dont have the data bag
+  void removeData(String key);
```

Comment:
`setData` is available under the transaction instance but there's no such `data` field, do we ignore it or set it in the `extra` field? on Android, we don't have the data field at all because of that

Reply:
on iOS:
transaction.setData(x, y)
transaction.contexts.trace.data

on .NET
transaction.setData(x, y)
transaction.extras[x] = y

Dart?

Reply:
it's done as the .NET API, if you set transaction.setData(x, y), it ends up in the extras field